### PR TITLE
Improve the TOTP regenerate flow

### PR DESCRIPTION
**Pull request by JayaShakthi97**
### Related Issues
- https://github.com/wso2/product-is/issues/14527

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Comment by JayaShakthi97** [comment]
Before:
Screenshot 2022-08-08 at 7 11 52 PM

After:
Screenshot 2022-08-08 at 5 16 53 PM

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx [hunk 1]
```diff
@@ -683,6 +685,13 @@ export const TOTPAuthenticator: React.FunctionComponent<TOTPProps> = (
             });
     };
 
+    /**
+     * Handle confirm QR code regeneration checkbox event
+     */
+    const handleConfirmRegenerateQRCode = (_: React.FormEvent<HTMLInputElement>, data: CheckboxProps): void => {
```
Comment (JayaShakthi97) [inline apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx:691]: Import `FormEvent` explicitly

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx [hunk 2]
```diff
@@ -729,9 +738,21 @@ export const TOTPAuthenticator: React.FunctionComponent<TOTPProps> = (
...
-                                    { t(translateKey + "modals.scan.regenerateWarning") }
+                                    { t(commonConfig.accountSecurityPage.mfa.totp.regenerateWarning) }
                                 </Message.Content>
                             </Message>
+                            {
+                                commonConfig.accountSecurityPage.mfa.totp.showRegenerateConfirmation
+                                    ? (
+                                        <Checkbox
+                                            className="mb-2"
+                                            label={ "Confirm regenerating a new QR code" }
```
Comment (dasuni-30) [inline apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx:749]: Can we add i18n strings here?

Comment (JayaShakthi97) [reply apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx]: Fixed with ad6160723f62e367fb76ac2aaed99ceaa4e9e986

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx (hunk 1 above)
Comment (JayaShakthi97) [reply apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx]: Fixed with ad6160723f62e367fb76ac2aaed99ceaa4e9e986

**Review by brionmario — APPROVED** [review]

File: apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx [hunk 3]
```diff
@@ -683,6 +685,13 @@ export const TOTPAuthenticator: React.FunctionComponent<TOTPProps> = (
             });
     };
 
+    /**
+     * Handle confirm QR code regeneration checkbox event
```
Comment (lashinijay) [inline apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx:689]: Shall we add a full stop at the end of the sentence? Check other places also.

Comment (JayaShakthi97) [reply apps/myaccount/src/components/multi-factor-authentication/authenticators/totp-authenticator.tsx:689]: Fixed with 8535eecb987fefadc2bc7058e3a9070e260baf10

**Pull request merged**